lnbits/static/js/components/extension-settings.js

```diff
@@ -0,0 +1,115 @@
+Vue.component('lnbits-extension-settings-form', {
+  name: 'lnbits-extension-settings-form',
+  props: ['options', 'name', 'cancel'],
+  computed: {
+    endpoint: function () {
+      return `/${this.name}/api/v1/settings?usr=${this.usr}`
+    }
+  },
+  methods: {
+    updateSettings: async function () {
+      try {
+        const {data} = await LNbits.api.request(
+          'PUT',
+          this.endpoint,
+          null,
+          this.settings
+        )
+        this.settings = data
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    },
+    getSettings: async function () {
+      try {
+        const {data} = await LNbits.api.request('GET', this.endpoint)
+        this.settings = data
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    },
+    resetSettings: async function () {
+      try {
+        await LNbits.api.request('DELETE', this.endpoint)
+        await this.getSettings()
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
```

Comment:
- a bit scary to delete without confirmation. Suggestion: add a `confirmDialog`
 
```suggestion
LNbits.utils
  .confirmDialog("Are you sure you want to reset the settings?")
  .onOk(async () => {
    try {
      await LNbits.api.request("DELETE", this.endpoint);
      await this.getSettings();
    } catch (error) {
      LNbits.utils.notifyApiError(error);
    }
  });
```